### PR TITLE
Subkey v2.0.2

### PR DESCRIPTION
**Pull request by mnaamani**
Subkey crate version was updated from 2.0.1 to 2.0.2,
Although docker image of older is available, latest on substrate repo is [v2.0.2](https://github.com/paritytech/substrate/blob/6a946fc36d68b89599d7ca1ab03803d10c78468c/bin/utils/subkey/Cargo.toml#L3)

This is to fix devops scripts that depend on subkey https://github.com/Joystream/joystream/runs/7199206582?check_suite_focus=true#step:8:108

cc: zeeshanakram3 FYI

Running new create aws AMI action here https://github.com/Joystream/joystream/actions/runs/2628777503

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202567022086904) by [Unito](https://www.unito.io)

### CODE REVIEW

File: devops/aws/roles/build-server/tasks/main.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   shell: curl https://getsubstrate.io -sSf | bash -s -- --fast
 
 - name: Install subkey
-  shell: cargo install --force subkey --git https://github.com/paritytech/substrate --version 2.0.1 --locked
+  shell: cargo install --force subkey --git https://github.com/paritytech/substrate --version 2.0.2 --locked
```
Comment (Lezek123) [inline devops/aws/roles/build-server/tasks/main.yml]: Should we allow minor version updates with:
```suggestion
  shell: cargo install --force subkey --git https://github.com/paritytech/substrate --version ^2.0.2 --locked
```
So we don't need to deal with this issue in the future?

**Review by Lezek123 — COMMENTED** [review]
Looks good,
There is also some commented-out code in `setup.sh` that could be updated as part of this PR (it seems related):
```bash
# Install substrate keychain tool - install doesn't seem to work lately.
# cargo install --force subkey --git https://github.com/paritytech/substrate --version 2.0.1 --locked
# You can use docker instead https://github.com/paritytech/substrate/tree/master/bin/utils/subkey#run-in-a-container
```

File: devops/aws/roles/build-server/tasks/main.yml (hunk 1 above)
Comment (mnaamani) [reply devops/aws/roles/build-server/tasks/main.yml]: Yeah that is a good idea, I'll confirm that it works and add it.

Comment (mnaamani) [reply devops/aws/roles/build-server/tasks/main.yml]: done in [d5a496d](https://github.com/Joystream/joystream/pull/4006/commits/d5a496dd7e38646f158e8aaf10bd796e20d9f6e4)

**Comment by mnaamani** [comment]
> Looks good, There is also some commented-out code in `setup.sh` that could be updated as part of this PR (it seems related):
> 
> ```shell
> # Install substrate keychain tool - install doesn't seem to work lately.
> # cargo install --force subkey --git https://github.com/paritytech/substrate --version 2.0.1 --locked
> # You can use docker instead https://github.com/paritytech/substrate/tree/master/bin/utils/subkey#run-in-a-container
> ```

Fixed in [e6fdfa1](https://github.com/Joystream/joystream/pull/4006/commits/e6fdfa1bb258e4336a8962037b727c408fd0b49b)

**Comment by mnaamani** [comment]
Also switched to using `joystream-node key` docker image in-place of subkey docker image where it is used as discussed out of band of this PR with @Lezek123

**Review by Lezek123 — APPROVED** [review]
LGTM!

**Pull request merged**